quisp/modules/ConnectionManager.cc

```diff
@@ -390,208 +393,147 @@ swap_table ConnectionManager::EntanglementSwappingConfig(int swapper_address, st
-
-  // }else{
-  //   error("this must not happen index must be positive");
-  // }
-
-  // if(swapper_address == 6&& left_partner==1&&right_partner==15){
-  //   error("good!");
-  // }
-  if (self_rqnic_type == QNIC_RP || self_lqnic_type == QNIC_RP || rqnic_type == QNIC_RP || lqnic_type == QNIC_RP) {
+  // left partner must be second
+  // right partner must be first
+  // TODO: detail description of this.
+  QNIC_id left_partner_qnic = qnics.at(left_partner_index).snd;
+  QNIC_id right_partner_qnic = qnics.at(right_partner_index).fst;
+
+  if (right_self_qnic.type == QNIC_RP || left_self_qnic.type == QNIC_RP || right_partner_qnic.type == QNIC_RP || left_partner_qnic.type == QNIC_RP) {
     error("MSM link not implemented");
   }
-  swap_setting.left_partner = left_partner;
-  swap_setting.lqnic_type = lqnic_type;
-  swap_setting.lqnic_index = lqnic_index;
-  swap_setting.lqnic_address = lqnic_address;
-  swap_setting.lres = num_resources;
 
-  swap_setting.right_partner = right_partner;
-  swap_setting.rqnic_type = rqnic_type;
-  swap_setting.rqnic_index = rqnic_index;
-  swap_setting.rqnic_address = rqnic_address;
-  swap_setting.rres = num_resources;
-
-  swap_setting.self_left_qnic_index = self_lqnic_index;
-  swap_setting.self_right_qnic_index = self_rqnic_index;
-  swap_setting.self_left_qnic_type = self_lqnic_type;
-  swap_setting.self_right_qnic_type = self_rqnic_type;
-
-  return swap_setting;
+  SwappingConfig config;
+  config.left_partner = left_partner;
+  config.lqnic_type = left_partner_qnic.type;
+  config.lqnic_index = left_partner_qnic.index;
+  config.lqnic_address = left_partner_qnic.address;
+  config.lres = num_resources;
+
+  config.right_partner = right_partner;
+  config.rqnic_type = right_partner_qnic.type;
+  config.rqnic_index = right_partner_qnic.index;
+  config.rqnic_address = right_partner_qnic.address;
+  config.rres = num_resources;
+
+  config.self_left_qnic_index = left_self_qnic.index;
+  config.self_right_qnic_index = right_self_qnic.index;
+  config.self_left_qnic_type = left_self_qnic.type;
+  config.self_right_qnic_type = right_self_qnic.type;
+  return config;
 }
 
 /**
- *  This function is called to handle the ConnectionSetupRequest at an "intermediate"
- *  node, one that is neither the initiator nor the responder.
- * \param pk pointer to the ConnectionSetupRequest packet itself
+ *  This function is called to handle the ConnectionSetupRequest at an intermediate or initiator node.
+ * \param req pointer to the ConnectionSetupRequest packet itself
  * \returns nothing
  **/
-void ConnectionManager::intermediate_alloc_req_handler(ConnectionSetupRequest *pk) {
-  int actual_dst = pk->getActual_destAddr();  // responder address
-  int actual_src = pk->getActual_srcAddr();  // initiator address (to get input qnic)
-  int local_qnic_address_to_actual_dst = routing_daemon->return_QNIC_address_to_destAddr(actual_dst);
+void ConnectionManager::relayRequestToNextHop(ConnectionSetupRequest *req) {
```

Comment:
I renamed `intermediate_alloc_req_handler` to this because the former name didn't describe it properly.
we should name the methods as describe what it does.